cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -583,6 +585,10 @@ ProgramOptions createProgramOptions(ProgramId programId, Map<String, String> use
                                                        profileId.getProfile(), profileId.getNamespace(),
                                                        programId.toString()), profileId);
     }
+    if (userProgramLaunchDisabled && profileId == ProfileId.NATIVE
+      && programId.getNamespaceId() != NamespaceId.SYSTEM) {
+      throw new RuntimeException("Launching user programs with Native profile is disabled.");
```

Comment:
this results in a 500 response code, I think a 403 would be more appropriate here

Reply:
Ah nice catch!

Reply:
<img width="1789" alt="image" src="https://user-images.githubusercontent.com/14131070/158869015-12d5ecd5-424b-414f-ab96-594155c99bc8.png">